opengate/engines.py

```diff
@@ -499,6 +647,64 @@ def initialize_optical_material_properties(self):
                         f"found in volume {vol.name} from file {self.physics_manager.optical_properties_file}."
                     )
 
+    def initialize_surface_material_properties(self):
+        """
+
+        This function deals with calling other functions and storing the
+        information required to create optical surfaces.
+
+        This information stored is used to create an optical surface (G4LogicalBorderSurface).
+        """
+
+        volume_surfaces_info = (
+            self.simulation_engine.simulation.physics_manager.volume_surfaces_info
+        )
+
+        for key, surfaces in volume_surfaces_info.items():
+            for surface in surfaces:
+                volume_1 = surface["volumes"][0]
+                volume_2 = surface["volumes"][1]
+
+                # Calls the function to get the physical volumes
+                # required to create optical surface (G4LogicalBorderSurface)
+                self.g4_physical_volumes = get_g4_physical_volumes(volume_1, volume_2)
```

Comment:
Better: get the GATE volume objects: 
vol1 = self.simulation_engine.get_volume(surface["volumes"][0])
vol1.g4_physical_volumes ...